actool/manifest.go

```diff
@@ -126,25 +170,47 @@ func patchManifest(im *schema.ImageManifest) error {
 		isolator = &types.Isolator{}
 		err := isolator.UnmarshalJSON([]byte(isolatorStr))
 		if err != nil {
-			return err
+			return fmt.Errorf("cannot parse capability %q: %v", patchCaps, err)
 		}
 		app.Isolators = append(app.Isolators, *isolator)
 	}
 
 	if patchMounts != "" {
-		app := im.App
-		if app == nil {
-			return fmt.Errorf("no app in the manifest")
-		}
 		mounts := strings.Split(patchMounts, ":")
 		for _, m := range mounts {
 			mountPoint, err := types.MountPointFromString(m)
 			if err != nil {
-				return fmt.Errorf("cannot parse mount point %q", m)
+				return fmt.Errorf("cannot parse mount point %q: %v", m, err)
 			}
 			app.MountPoints = append(app.MountPoints, *mountPoint)
 		}
 	}
+
+	if patchPorts != "" {
+		ports := strings.Split(patchPorts, ":")
+		for _, p := range ports {
+			port, err := types.PortFromString(p)
+			if err != nil {
+				return fmt.Errorf("cannot parse port %q: %v", p, err)
+			}
+			app.Ports = append(app.Ports, *port)
+		}
+	}
+
+	if patchIsolators != "" {
+		isolators := strings.Split(patchIsolators, ":")
+		for _, is := range isolators {
+			isolatorStr, err := isolatorStrFromString(is)
+			if err != nil {
+				return fmt.Errorf("cannot parse isolator %q: %v", is, err)
```

Comment:
"cannot parse isolator: isolator x is not supported for patching" is a bit awkward; maybe parsing should succeed but then you should test the type here?